src/v/cluster/remote_topic_configuration_source.cc

```diff
@@ -49,7 +50,65 @@ remote_topic_configuration_source::set_remote_properties_in_config(
           key,
           bucket,
           res);
+        co_return std::make_tuple(errc::topic_operation_error, key);
+    }
+    co_return std::make_tuple(errc::success, key);
+}
+
+ss::future<errc>
+remote_topic_configuration_source::set_remote_properties_in_config(
+  custom_assignable_topic_configuration& cfg,
+  const s3::bucket_name& bucket,
+  ss::abort_source& as) {
+    cloud_storage::topic_manifest manifest;
+
+    auto [res, key] = co_await download_topic_manifest(
+      _remote, cfg, manifest, bucket, as);
+    if (res != errc::success) {
+        co_return res;
+    }
+
+    if (!manifest.get_topic_config()) {
+        vlog(
+          clusterlog.warn,
+          "Topic manifest {} doesn't contain topic config",
+          key);
         co_return errc::topic_operation_error;
+    } else {
+        cfg.cfg.properties.remote_topic_properties = remote_topic_properties(
+          manifest.get_revision(),
+          manifest.get_topic_config()->partition_count);
+    }
+    co_return errc::success;
+}
+
+/// If property is set in source apply it to target.
+static void apply_retention_defaults(
+  topic_properties& target, const topic_properties& source) {
+    // If the retention properties are not set explicitly by the command we
+    // should apply them from topic_manifest.
+    if (!target.cleanup_policy_bitflags) {
+        target.cleanup_policy_bitflags = source.cleanup_policy_bitflags;
+    }
+    if (!target.retention_bytes.has_value()) {
+        target.retention_bytes = source.retention_bytes;
+    }
+    if (!target.retention_duration.has_value()) {
+        target.retention_duration = source.retention_duration;
+    }
+}
+
+ss::future<errc>
+remote_topic_configuration_source::set_recovered_topic_properties(
```

Comment:
looks like this method does the same as set_remote_properties_in_config +  sets partition_count and calls `apply_retention_defaults`. Maybe they can share some logic?

Reply:
I already extracted `download_topic_manifest` method.